### PR TITLE
Pylons now count only groundside marines, xenos get major boost to evo speed for 3 minutes after hijack

### PR DESCRIPTION
**Pull request by ihatethisengine**
# About the pull request
Pylons give larva only up to 50% of groundside (weighted) marines, instead of 40% of all humans.
Xenos get major boost to evo speed for 3 minutes after hijack, also hijacking removes ovi cd.

# Explain why it's good for the game
Larva surge works this way (but for shipside), and pylons were kinda meant to work this way too. Current implementation encourages xenos to delay hijack as long as possible to get the maximum larva, get evo etc. which is not good. It shouldn't break balance too much, you should be able to defeat marines if you have 40% of their force.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl: ihatethisengine
balance: Pylons give larva only up to 50% of groundside (weighted) marines, instead of 40% of all humans.
balance: Xenos get major boost to evo speed for 3 minutes after hijack. Hijacking resets ovi cooldown.
/:cl:

### CODE REVIEW

File: code/modules/cm_aliens/structures/special/pylon_core.dm
Comment (fira) [inline code/modules/cm_aliens/structures/special/pylon_core.dm:208]: that's the opposite???

**Review by fira — CHANGES_REQUESTED** [review]

File: code/modules/cm_aliens/structures/special/pylon_core.dm
Comment (ihatethisengine) [reply code/modules/cm_aliens/structures/special/pylon_core.dm:208]: Happens to the best, I wanted to invert is shipside level, but found this proc

**Comment by ihatethisengine** [comment]
Why dont it work :sob:

**Comment by Birdtalon** [comment]
I'd say this needs balance approval since xeno endgame relies largely on pylon larva at the moment to siege and then execute a successful hijack. Based on my gut feeling this PR could reduce the larva gain by anything from 20-50% some rounds which I would say is not insignificant.

**Comment by ihatethisengine** [comment]
We can up the number to 50%. Theoretically all groundside marines (except doctors) should have weight of 1, so it shouldn't be an issue. Realistically if you have 20 xenos vs 40 marines it's xeno major, especially since you get new xenos when old xenos die.

**Comment by GoldenDarkness55** [comment]
less numbers = you're even more encouraged to wait for evo.
Just add some kinda check for queen ovi on DS that speeds evo up tenfold for a minute or smth

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
Little bit out of scope, but could you please add something that increases evo speed for a period after the queen launches the DS?

Lgtm otherwise

**Comment by ihatethisengine** [comment]
I'll try.

**Comment by ihatethisengine** [comment]
Done

File: code/modules/cm_aliens/structures/special/pylon_core.dm
Comment (Drulikar) [inline code/modules/cm_aliens/structures/special/pylon_core.dm:185]: Revert this incorrect autodoc change moving it somereason to ENDGAME_LARVA_CAP_MULTIPLIER.

File: code/modules/cm_aliens/structures/special/pylon_core.dm
Comment (Drulikar) [inline code/modules/cm_aliens/structures/special/pylon_core.dm:210]: Since you cited "Current implementation encourages xenos to delay hijack as long as possible to get the maximum larva" you'd have more effect if hitting this cap in some way was communicated to the hive. I'm not really suggesting it should have a text announcement (if it does it should only do it infrequently and not be spammy); but likely would be good to think of maybe a way for this to be indicated in the hive status.

File: code/modules/mob/living/carbon/xenomorph/hive_status.dm
Comment (Drulikar) [inline code/modules/mob/living/carbon/xenomorph/hive_status.dm:1289]: Is it possible for this subsystem to not be present?

I mean the check is okay, but it seems unnecessary? I'm more concerned what situation won't have the subsystem?

File: code/modules/shuttle/computers/dropship_computer.dm
Comment (Drulikar) [inline code/modules/shuttle/computers/dropship_computer.dm:304]: This should be a define somewhere rather than hard coded in this proc.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/modules/shuttle/computers/dropship_computer.dm
Comment (Drulikar) [inline code/modules/shuttle/computers/dropship_computer.dm:307]: It would be better to reduce by the cooldown amount for better future proofing. E.g. maybe ovi cooldown nerfed to 10 minutes, then this will only reduce the time an amount not reset it.
```suggestion
		ovi_ability.reduce_cooldown(ovi_ability.xeno_cooldown)
```
Untested suggestion, but should be correct.

File: code/modules/shuttle/computers/dropship_computer.dm
Comment (Drulikar) [inline code/modules/shuttle/computers/dropship_computer.dm:308]: The 3 MINUTES should also be a define.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/modules/cm_aliens/structures/special/pylon_core.dm
Comment (ihatethisengine) [reply code/modules/cm_aliens/structures/special/pylon_core.dm:185]: oops it was an accident

File: code/modules/mob/living/carbon/xenomorph/hive_status.dm
Comment (ihatethisengine) [reply code/modules/mob/living/carbon/xenomorph/hive_status.dm:1289]: I just copypasted it from another place, not sure

File: code/modules/cm_aliens/structures/special/pylon_core.dm
Comment (ihatethisengine) [reply code/modules/cm_aliens/structures/special/pylon_core.dm:213]: I added messages, but I am afraid it still might be spammy. Especially because pylons provide larvae separately, so time between procs may differ significantly. Any better ideas?

Comment (ihatethisengine) [reply code/modules/cm_aliens/structures/special/pylon_core.dm:213]: I probably can add some kind of timer that will check if hit_larva_limit is still FALSE after few minutes and send the message only then... Seems kinda lame, but cannot think of a better solution

Comment (ihatethisengine) [reply code/modules/cm_aliens/structures/special/pylon_core.dm:213]: and still there can be a situation, when I send the message and the hit_larva instantly becomes true and sends another message =\\\

Comment (ihatethisengine) [reply code/modules/cm_aliens/structures/special/pylon_core.dm:213]: It's almost like I need to use one timer for both messages and queue them correctly, hmmm

Comment (Drulikar) [reply code/modules/cm_aliens/structures/special/pylon_core.dm:213]: As I mentioned, I think it would be better to be information in the hive_status rather than messages:
> I'm not really suggesting it should have a text announcement (if it does it should only do it infrequently and not be spammy); but likely would be good to think of maybe a way for this to be indicated in the hive status.

File: code/modules/shuttle/computers/dropship_computer.dm
Comment (Drulikar) [inline code/modules/shuttle/computers/dropship_computer.dm:267]: Move these to https://github.com/cmss13-devs/cmss13/blob/a730d2b5cbf69388d919396e7a94b03cd5dcb8e9/code/__DEFINES/xeno.dm#L171-L178 with autodoc comments to be with the rest of most tweakable settings for xenos. Or a different define file if you find one more appropriate.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/modules/cm_aliens/structures/special/pylon_core.dm
Comment (Drulikar) [reply code/modules/cm_aliens/structures/special/pylon_core.dm:213]: E.g. here: 
image

Comment (ihatethisengine) [reply code/modules/cm_aliens/structures/special/pylon_core.dm:213]: yeah I can use tgui I code via VM. I don't think anyone would check hivestatus to know pylon status though

Comment (Drulikar) [reply code/modules/cm_aliens/structures/special/pylon_core.dm:213]: Almost every xeno uses this UI. And if theres new text near burrowed larva they will see it, because people look at that number all the time.

Comment (ihatethisengine) [reply code/modules/cm_aliens/structures/special/pylon_core.dm:213]: Alright I figure something out a bit later

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/__DEFINES/xeno.dm
Comment (Drulikar) [inline code/__DEFINES/xeno.dm:178]: Revert

File: code/modules/cm_aliens/structures/special/pylon_core.dm
Comment (Drulikar) [inline code/modules/cm_aliens/structures/special/pylon_core.dm:181]: Unnecessary, but doesn't hurt either. You only need to TIMER_NO_HASH_WAIT if the duration changes, and in this case it can't change.

File: code/modules/cm_aliens/structures/special/pylon_core.dm
Comment (Drulikar) [inline code/modules/cm_aliens/structures/special/pylon_core.dm:186]: Revert

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/__DEFINES/xeno.dm
Comment (fira) [inline code/__DEFINES/xeno.dm:184]: use ( ) around time defines to avoid people tripping up on math operators - like XENO_PYLON_DESTRUCTION_DELAY above

**Review by fira — APPROVED** [review]
Code LGTM

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/modules/cm_aliens/structures/special/pylon_core.dm
Comment (Drulikar) [inline code/modules/cm_aliens/structures/special/pylon_core.dm:191]: I believe you have this reversed from what you want.

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Review by Drulikar — APPROVED** [review]
Code changes look fine.

**Pull request merged**
